src/index.js

```diff
@@ -0,0 +1,903 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-warning-comments */
+/* eslint-disable valid-jsdoc */
+
+'use strict'
+
+const Pubsub = require('libp2p-pubsub')
+const pull = require('pull-stream')
+const lp = require('pull-length-prefixed')
+const nextTick = require('async/nextTick')
+
+const MessageCache = require('./messageCache').MessageCache
+const utils = require('./utils')
+const assert = require('assert')
+
+const RPC = require('./message').rpc
```

Comment:
The way this `require` is being made + how it is used is causing `subscribe` and`publish` to fail.

```suggestion
const { rpc } = require('./message')
```

Then for using this, you should use `rpc.RPC.encode()` and `rpc.RPC.decode()`

Basically, when a peer receives a message from `subscribe`, the `_processConnection` is failing because of the current `RPC.encode()` function does not exist. Also, the messages sent like the `_sendGraft` should also have the extra property `RPC`. 

If you prefer, you can also do the require in a way that you don't need to change the rest of the code.